### ISSUE TITLE
Configuration confusion bootstrap.servers vs. zookeeper.connect

### ISSUE THREAD

**Issue by tweise** [confluentinc/kafka-rest#346]
The template properties only contain zookeeper.connect and in theory that should be sufficient to discover the brokers. In reality, while this works for the producer, the consumer will fail to connect. It would be nice to either document the need for bootstrap.servers or derive it from zookeeper.connect when not present. Rest proxy v3.3.0

**Comment by jomach** [comment]
Yeah, I agreed. The docs are not very helpful.  And why is not possible to change the jvm settings without changing code ? export from JVM_OPTS or KAFKA_OPTS

**Comment by mageshn** [comment]
@tweise we already derive bootstrap.servers from zookeeper.connect when its not present. Having said that in future releases, we will have bootstrap.servers as the default config specified in the config.

@jomach  I don't think I completely understand your concern about changing code for JVM settings. You can just export the JVM settings and you should be good to go.

**Comment by tweise** [comment]
@mageshn that's not what I observed. I started with just setting `zookeeper.connect` as that's what I saw in the template and then had to also add `bootstrap.servers` when the consumer (not the producer) failed.

**Comment by mageshn** [comment]
@tweise would be great to see the errors when you don't have bootstrap brokers specified.

**Comment by tweise** [comment]
it was trying to connect to localhost:9092

**Comment by Dutzu** [comment]
I have the exact same problem.

Here is a log sample when a consumer is created:
``` 
[2017-10-31 12:17:29,116] INFO ConsumerConfig values:
kafka-rest1_1       |   auto.commit.interval.ms = 5000
kafka-rest1_1       |   auto.offset.reset = earliest
kafka-rest1_1       |   bootstrap.servers = [PLAINTEXT://localhost:9092]
```

And here is a producer config entry:
```
kafka-rest1_1       | [2017-10-31 09:49:10,697] INFO ProducerConfig values:
kafka-rest1_1       |   acks = 1
kafka-rest1_1       |   batch.size = 16384
kafka-rest1_1       |   block.on.buffer.full = false
kafka-rest1_1       |   bootstrap.servers = [PLAINTEXT://10.80.7.15:19092, PLAINTEXT://10.80.7.15:29092]
```

These are the first log lines after starting the process:
```
kafka-rest1_1       | Zookeeper: zookeeper1:12181,zookeeper2:22181,zookeeper3:32181
kafka-rest1_1       | Kafka rest listeners: http://0.0.0.0:8082
kafka-rest1_1       | Schema registry: http://schema-registry1:19081,http://schema-registry2:29081
kafka-rest1_1       | Hostname: kafka-rest1
kafka-rest1_1       | Instance ID: kafka-rest1
kafka-rest1_1       | [2017-10-31 09:49:09,573] INFO KafkaRestConfig values:
kafka-rest1_1       |   metric.reporters = []
kafka-rest1_1       |   client.security.protocol = PLAINTEXT
kafka-rest1_1       |   bootstrap.servers = PLAINTEXT://localhost:9092
```

So from what i can see, even though it has succesfully determined thebootstrap servers via zookeeper for the producers, it doesn't do the same thing for the consumers.

What would be the correct approach in this case?

**Comment by ujlbu4** [comment]
I have the same problem.

And as @tweise wrote, I just added `bootstrap.servers` to launch params:
```
docker run -d \
  --net=host \
  --name=kafka-rest \
  -e KAFKA_REST_ZOOKEEPER_CONNECT=kafka1.example.com:2181,kafka2.example.com:2181,kafka3.example.com:2181 \
  -e KAFKA_REST_BOOTSTRAP_SERVERS=kafka1.example.com:9092,kafka2.example.com:9092,kafka3.example.com:9092 \
  -e KAFKA_REST_HOST_NAME=proxy-kafka.example.com \
  confluentinc/cp-kafka-rest:3.3.0
```

This workaround works for me, but I expecting that zookeeper params should be enough. Especially when producers works fine.

**Comment by mageshn** [comment]
I think its a reasonable workaround to use the boot strap broker to get it working  because in long run we would like to completely remove ZK dependency from Rest Proxy.

**Comment by ujlbu4** [comment]
Maybe it will be better to update configuration documentation: https://docs.confluent.io/current/cp-docker-images/docs/configuration.html#kafka-rest-proxy ?

cause there are no such a note about bootstrap param

**Comment by mageshn** [comment]
@ujlbu4 thanks for your feedback.. we will do that.
